src/proper.erl

```diff
@@ -1774,12 +1751,14 @@ shrink(Shrunk, [ImmInstance | Rest], {_Type,Prop}, Reason,
 	   Shrinks, ShrinksLeft, init, Opts);
 shrink(Shrunk, [RawImmInstance | Rest] = TestTail, {Type,Prop} = StrTest, Reason,
        Shrinks, ShrinksLeft, State, Opts) ->
-    ImmInstance = case proper_types:find_prop(user_nf, Type) of
-		      {ok, _} ->
-			  proper_gen:clean_instance(RawImmInstance);
-		      error ->
-			  RawImmInstance
-		  end,
+    ImmInstance = case get('$user_nf') of
+                    true ->
+                      case proper_types:safe_is_instance(RawImmInstance, Type) of
+                        false -> proper_gen:clean_instance(RawImmInstance);
+                        _ -> RawImmInstance
+                      end;
+                    _ -> RawImmInstance
```

Comment:
Similarly to the above, the `_` here should read `false` for avoiding this case matching if `$user_nf` is not set.

Reply:
This could be `undefined` if `$user_nf` is not set at all.